babble-frontend/src/App.js

```diff
@@ -1,21 +1,297 @@
+
+  // authentication calls
+  const createUser = useCallback( async (body) => {
+    try {
+      const result = await axios.post(`/enter/signup`, body);
+      console.log(result);
+      alert("Account created! Please go to login.");
+      return true;
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
+  });
+
+  const loginUser = useCallback( async ({body, callback}) => {
+    try {
+      const result = await axios.post(`/enter/login`, body);
+      console.log(result);
+      localStorage.setItem(
+        "AuthToken",
+        JSON.stringify(result.data),
+      );
+      token = localStorage.getItem("AuthToken");
+      console.log(token);
+      authAxios = axios.create({
+        baseURL: apiURL,
+        headers: {
+          authentication: token
+        }
+      });
+      setLoginInfo(body);
+      localStorage.setItem(
+        "email",
+        body.email
+      );
+      localStorage.setItem(
+        "pswd",
+        body.password
+      );//not secure at all but im v lazy rn
+      callback();
+    } catch (err) {
+      console.log(err);
+      alert("Invalid Login")
+    }
+  });
+
+  const logoutUser = useCallback( async () => {
+    try {
+      const result = await authAxios.post(`/user/logout`);
+      return true;
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
+  });
+
+  // user info calls
+  const deleteUser = useCallback( async () => {
+    try {
+      const body = {
+        email: localStorage.getItem("email"),
+        password: localStorage.getItem("pswd"),
+      };
+      console.log(body);
+      const delAxios = axios.create({
+        baseURL: apiURL,
+        headers: {
+          authentication: localStorage.getItem("AuthToken"),
+          email: body.email,
+          password: body.password,
+        },
+      });
+      const result = await delAxios.delete(`/user/delete-account`);
+      if(result.status===200){
+        alert("Deleted!")
+      }else{
+        alert("Deletion invalid")
+      }
+      return logoutUser();
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
+  });
+
+  const changePassword = useCallback( async (newP) => {
+    try {
+      const body = {
+        email: localStorage.getItem("email"),
+        currentPassword: localStorage.getItem("pswd"),
+        newPassword: newP
+      }
+      const result = await authAxios.post(`/user/change-password`, body);
+      console.log(result);
+      return logoutUser();
+    } catch (err) {
+      console.log(err);
+    }
+  });
+
+  const changeEmail = useCallback( async (newE) => {
+    try {
+      const body = {
+        currentEmail: localStorage.getItem("email"),
+        newEmail: newE,
+        password: localStorage.getItem("pswd"),
+      }
+      const result = await authAxios.post(`/user/change-email`, body);
+      return logoutUser();
+    } catch (err) {
+      console.log(err);
+      return false;
+    }
+  });
```

Comment:
We still need to store the email and password for display purposes on the frontend on the profile page anyway, I would usually just use redux since localstorage is permanent, but I dont want to bloat our project. I'm looking into session storage. You will need to talk to nicole about implementing a different UI to edit things.